lib/order.rb

```diff
@@ -0,0 +1,97 @@
+require_relative 'customer.rb'
+require 'csv'
+
+
+class Order
+  # read-only ID number
+  attr_reader :id, :products, :customer, :fulfillment_status
+
+  
+  def initialize(id, products_hash, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products_hash
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+   
+    # raises ArgumentError if condition isn't met
+    unless [:pending, :paid, :processing, :shipped, :complete].include?(fulfillment_status)
+      raise ArgumentError, "Fulfillment status must be valid."
+      @fulfillment_status = fulfillment_status
+    end
+  end
+
+  # if hash has elements in it, calculate total. Else total = 0 
+  def total
+    if @products.length > 0
+      expected_total = (@products.values.map.reduce(:+)) + (@products.values.map.reduce(:+) *(0.075)).round(2)
+    else
+      expected_total = 0
+    end
+      return expected_total
+  end
+
+  # add elements to the products hash
+  def add_product(product_name, price)
+    if @products.include?(product_name)
+      raise ArgumentError, "This product already exists in the inventory."
+    else
+    end
```

Comment:
If you have this kind of syntax (with an empty `else` block), feel free to just let it be `if ... end` without the `else`!